spras/evaluation.py

```diff
@@ -0,0 +1,46 @@
+import os
+import pickle as pkl
+import warnings
+
+import pandas as pd
+
+class Evaluation:
+   
+    def __init__(self, gold_standard_dict):
+        self.label = None
+        self.node_table = None
+        # self.edge_table = None TODO: later iteration
+        self.load_files_from_dict(gold_standard_dict)
+        return
+    
+    def to_file(self, file_name):
+        """
+        Saves dataset object to pickle file
+        """
+        with open(file_name, "wb") as f:
+            pkl.dump(self, f)
+
+    @classmethod
+    def from_file(cls, file_name):
+        """
+        Loads dataset object from a pickle file.
+        Usage: dataset = Dataset.from_file(pickle_file)
+        """
+        with open(file_name, "rb") as f:
+            return pkl.load(f)
+
+    def load_files_from_dict(self, gold_standard_dict):
+         
+        self.label = gold_standard_dict["label"]
+        node_data_files = gold_standard_dict["node_files"]
+        data_loc = gold_standard_dict["data_dir"]
+
+        single_node_table = pd.read_table(os.path.join(data_loc, node_file))
+        self.node_table = single_node_table
+
+        # self.node_table = pd.DataFrame(node_set, columns=[self.NODE_ID])
+        # for loop? and read in node dataset into a pandas df
+
+def percision_recall():
+    None
+    # https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html
```

Comment:
This is the best metric to use if we have node ranks. Should we start with the simplest thing and evaluate the entire set of nodes on the predicted pathway because we usually don't have node ranks? Then we could use metrics like precision, recall, F1, and MCC.

Reply:
I'll try using precision first

Reply:
Noting here that @sumedhars is also prototying an implementation of node ranking and precision-recall curves in her project fairly soon. We can coordinate because if it works for her, she can contribute the code here or in a follow up pull request.